lib/chef_zero/server.rb

```diff
@@ -164,10 +164,10 @@ def port
     #
     def url
       sch = @options[:ssl] ? 'https' : 'http'
-      @url ||= if @options[:host].include?(':')
-                 URI("#{sch}://[#{@options[:host]}]:#{port}").to_s
+      @url ||= if @options[:host].first.include?(':')
+                 URI("#{sch}://[#{@options[:host].first}]:#{port}").to_s
                else
-                 URI("#{sch}://#{@options[:host]}:#{port}").to_s
+                 URI("#{sch}://#{@options[:host].first}:#{port}").to_s
```

Comment:
In fact, @danielsdeleo - this code seems to not be used much. It's only used in rspecs and `inspect` methods. A later diff can probably nuke that method all together. We build real URLS for REST requests based on the hostname used to talk to us:

```
rest_request.rb:      @base_uri ||= "#{env['rack.url_scheme']}://#{env['HTTP_HOST']}#{env['SCRIPT_NAME']}"
```